lib/Sema/TypeCheckConcurrency.cpp

```diff
@@ -385,6 +385,7 @@ ActorIsolationRestriction ActorIsolationRestriction::forDeclaration(
       if (func->isAsyncContext())
         isAccessibleAcrossActors = true;
 
+      // FIXME: move diagnosis out of this function entirely (!)
```

Comment:
I'll do this cleanup shortly. but separately would be better so we have the new function in so @kavon can start poking at implementing it.